### PR TITLE
add custom easyblock for CRISPR-DAV

### PR DESCRIPTION
**Pull request by deniskristak**
needed to create easyblock due to complex source file changes

### CODE REVIEW

File: easybuild/easyblocks/c/crisprdav.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+EasyBuild support for building and installing Metagenome-Atlas, implemented as an easyblock.
+
+@author: Denis Kristak (INUITS)
+"""
+import os
+from easybuild.tools.filetools import write_file
+from easybuild.easyblocks.generic.binary import Binary
+
+
+class EB_crisprdav(Binary):
```
Comment (boegel) [inline easybuild/easyblocks/c/crisprdav.py:35]: please rename to `EB_CRISP_minus_DAV` (see also https://github.com/easybuilders/easybuild-easyconfigs/pull/13139)

File: easybuild/easyblocks/c/crisprdav.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        write_file(config_file, CONFIG_FILE_TEXT_FORMATTED)
+        write_file(example_config_file, CONFIG_FILE_TEXT_FORMATTED)
+        # also used for sanity checking with Examples/ folder
+        write_file(example_fastq_list_file, FASTQ_LIST_FILE_TEXT_FORMATTED)
+
+
+# obtained from
+# https://github.com/pinetree1/crispr-dav/blob/master/conf.txt
+# see the original file for detailed commentary
+CONFIG_FILE_TEXT = """
```
Comment (boegel) [inline easybuild/easyblocks/c/crisprdav.py:69]: constants should be defined above the `class` definition

File: easybuild/easyblocks/c/crisprdav.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+# change this part to contain paths to your genomes
+[genomex]
+ref_fasta = {crisprdav_installdir_ex}/genome/genomex.fa
+bwa_idx = {crisprdav_installdir_ex}/genome/genomex.fa
+refGene = {crisprdav_installdir_ex}/genome/refgenex.txt
+
+[app]
+abra = {abra2_dir}/abra2-2.23.jar
```
Comment (boegel) [inline easybuild/easyblocks/c/crisprdav.py:77]: Wouldn't it be better to copy the `conf.txt`, and do a couple of search-and-replace operations on that?

Shipping a copy of the `conf.txt` in this easyblock makes it a maintenance problem going forward...

File: easybuild/easyblocks/c/crisprdav.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+realign_flag = Y
+min_mapq = 20
+wing_length = 100
+high_res = 0
+parallel_env = orte
+cores_per_job = 12
+"""
+
+FASTQ_LIST_FILE_TEXT = r"""
+sample1	{crisprdav_installdir_ex}/rawfastq/sample1_R1.fastq.gz	{crisprdav_installdir_ex}/rawfastq/sample1_R2.fastq.gz
```
Comment (boegel) [inline easybuild/easyblocks/c/crisprdav.py:96]: Isn't it better to *generate* this file?

File: easybuild/easyblocks/c/crisprdav.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+
+class EB_crisprdav(Binary):
+    """
+    Support for building/installing crispr-dav.
+    """
+
+    def post_install_step(self):
+        """Create config.txt files"""
+        crisprdav_installdir_ex = os.path.join(self.installdir, 'Examples/example1/')
```
Comment (boegel) [inline easybuild/easyblocks/c/crisprdav.py:42]: please avoid using `/` in `os.path.join` (same below):

```python
crisprdav_installdir_ex = os.path.join(self.installdir, 'Examples', 'example1')
```

File: easybuild/easyblocks/c/crisprdav.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        abra2_dir = os.environ['EBROOTABRA2']
+        prinseq_dir = os.environ['EBROOTPRINSEQ']
+        flash_dir = os.environ['EBROOTFLASH']
```
Comment (boegel) [inline easybuild/easyblocks/c/crisprdav.py:45]: It's better to use the `get_software_root` function here, and you should also *check* that the result is not `None`, which verifies that these are indeed included as dependencies.

File: easybuild/easyblocks/c/crisprdav.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Metagenome-Atlas, implemented as an easyblock.
```
Comment (boegel) [inline easybuild/easyblocks/c/crisprdav.py:26]: Please fix, this easyblock is for `CRISPR-DAV`, not for `Metagenome-Atlas`

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by verdurin** [comment]
@deniskristak I tried building the associated easyconfig and saw this error:
```
eb --from-pr=13139 --robot --debug --upload-test-report --include-easyblocks-from-pr 2487
== Temporary log file in case of crash /tmp/eb-jdvm00ah/easybuild-q0xclof4.log
== easyblock crispr_dav.py included from PR #2487
ERROR: Failed to process easyconfig /tmp/eb-jdvm00ah/files_pr13139/c/crispr-dav/crispr-dav-2.3.4-foss-2020b.eb: Failed to obtain class for EB_crisprdav easyblock (not available?): No module named 'easybuild.easyblocks.crisprdav'
```

**Comment by deniskristak** [comment]
@verdurin correct, i forgot to remove an easyconfig with an the old name. Should be working now

**Comment by boegel** [comment]
close/re-open to wake up GitHub Actions...

**Review by boegel — APPROVED** [review]

**Pull request merged**